File: easy/problems/0401-binary-watch/solution.py
def readBinaryWatch(num):
    """
    Display times on a binary watch based on number of lit LEDs.

    Args:
        num: int - number of lit LEDs

    Returns:
        List[str] - all possible times
    """
    result = []

    for h in range(12):
        for m in range(60):
            if bin(h).count('1') + bin(m).count('1') == num:
                result.append(f"{h}:{m:02d}")

    return result

File: easy/problems/0401-binary-watch/solution.ts
function readBinaryWatch(num: number): string[] {
    const result: string[] = [];

    for (let h = 0; h < 12; h++) {
        for (let m = 0; m < 60; m++) {
            const bitCount = (h.toString(2).match(/1/g) || []).length +
                           (m.toString(2).match(/1/g) || []).length;
            if (bitCount === num) {
                result.push(`${h}:${String(m).padStart(2, '0')}`);
            }
        }
    }

    return result;
}
